src/index.ts

```diff
@@ -368,758 +145,594 @@ export default class OneSignal {
+     */
+    export function setLogLevel(nsLogLevel: LogLevel) {
+      if (!isNativeModuleLoaded(RNOneSignal)) return;
+
+      RNOneSignal.setLogLevel(nsLogLevel);
     }
 
-    RNOneSignal.setLocationShared(shared);
-  }
+    /**
+     * Enable logging to help debug if you run into an issue setting up OneSignal.
+     * @param {LogLevel} visualLogLevel - Sets the logging level to show as alert dialogs.
+     */
+    export function setAlertLevel(visualLogLevel: LogLevel) {
+      if (!isNativeModuleLoaded(RNOneSignal)) return;
 
-  /**
-   * Prompts the user for location permissions to allow geotagging from the OneSignal dashboard.
-   * @returns void
-   */
-  static promptLocation(): void {
-    if (!isNativeModuleLoaded(RNOneSignal)) {
-      return;
+      RNOneSignal.setAlertLevel(visualLogLevel);
     }
-
-    //Supported in both iOS & Android
-    RNOneSignal.promptLocation();
   }
 
-  /* D E V I C E  I N F O */
+  export namespace User {
+    export namespace PushSubscription {
+      /** Add a callback that fires when the OneSignal subscription state changes. */
+      export function addChangeHandler(
+        observer: (event: ChangeEvent<SubscriptionChange>) => void,
+      ) {
+        if (!isNativeModuleLoaded(RNOneSignal)) return;
 
-  /**
-   * Gets the device state.
-   * This method returns a "snapshot" of the device state for when it was called.
-   * @returns Promise<DeviceState | null>
-   */
-  static async getDeviceState(): Promise<DeviceState | null> {
-    if (!isNativeModuleLoaded(RNOneSignal)) {
-      return Promise.resolve(null);
-    }
-    const deviceState = await RNOneSignal.getDeviceState();
+        isValidCallback(observer);
+        RNOneSignal.addPushSubscriptionChangeHandler();
+        eventManager.addEventHandler<SubscriptionChange>(
+          SUBSCRIPTION_CHANGED,
+          observer,
+        );
+      }
 
-    if (Platform.OS === 'android') {
-      deviceState.hasNotificationPermission =
-        deviceState.areNotificationsEnabled;
-      delete deviceState.areNotificationsEnabled;
-    }
+      /** Clears current subscription observers. */
+      export function removeChangeHandler() {
+        if (!isNativeModuleLoaded(RNOneSignal)) return;
 
-    return deviceState;
-  }
+        RNOneSignal.removePushSubscriptionChangeHandler();
+        eventManager.clearEventHandler(SUBSCRIPTION_CHANGED);
+      }
 
-  /**
-   * Allows you to set the app defined language with the OneSignal SDK.
-   * @param  {string} language
-   * @param  {(success:object) => void} onSuccess
-   * @param  {(failure:object) => void} onFailure
-   * @returns void
-   */
-  static setLanguage(
-    language: string,
-    onSuccess?: (success: object) => void,
-    onFailure?: (failure: object) => void,
-  ): void {
-    if (!isNativeModuleLoaded(RNOneSignal)) {
-      return;
-    }
+      /** The readonly push subscription ID */
+      export function getPushSubscriptionId(): Promise<string> {
```

Comment:
in cordova/flutter we kept push subscription id/token/optedIn states within the wrapper, and used the change handler to ensure the wrapper state was consistent to the native state.  Do you think it's possible to keep the react SDK consistent with the others?

Reply:
Done